### PR TITLE
Adds a heavily-damaged ripley as a lavaland ruin

### PR DESCRIPTION
**Pull request by ChangelingRain**
It starts off at low health and with low cell charge, but randomly has a drill, plasmacutter, or diamond drill, in addition to an ore box, clamp, and mining scanner.

It's basically like the old random room on the asteroid with a ripley in it, except it's harder to find since you don't have turf vision.

### CODE REVIEW

**Comment by Xhuis** [comment]
Any screenshots handy?

**Comment by ChangelingRain** [comment]
It's basically a 5x5 block of rock with a few open turfs and a ripley in the center.

File: code/game/mecha/working/ripley.dm [hunk 1]
```diff
@@ -91,19 +91,21 @@
...
+/obj/mecha/working/ripley/mining/Initialize()
+	. = ..()
+	if(cell)
+		cell.charge = Floor(cell.charge * 0.25) //Starts at very low charge
+	if(prob(70)) //Maybe add a drill
+		if(prob(15)) //Possible diamond drill... Feeling lucky?
+			var/obj/item/mecha_parts/mecha_equipment/drill/diamonddrill/D = new /obj/item/mecha_parts/mecha_equipment/drill/diamonddrill
+			D.attach(src)
+		else
+			var/obj/item/mecha_parts/mecha_equipment/drill/D = new /obj/item/mecha_parts/mecha_equipment/drill
```
Comment (Cyberboss) [inline code/game/mecha/working/ripley.dm]: just `= new` for less redundancy

File: code/game/mecha/working/ripley.dm [hunk 2]
```diff
@@ -91,19 +91,21 @@
...
-	else
-		var/obj/item/mecha_parts/mecha_equipment/drill/D = new /obj/item/mecha_parts/mecha_equipment/drill
-		D.attach(src)
+/obj/mecha/working/ripley/mining/Initialize()
+	. = ..()
+	if(cell)
+		cell.charge = Floor(cell.charge * 0.25) //Starts at very low charge
+	if(prob(70)) //Maybe add a drill
+		if(prob(15)) //Possible diamond drill... Feeling lucky?
+			var/obj/item/mecha_parts/mecha_equipment/drill/diamonddrill/D = new /obj/item/mecha_parts/mecha_equipment/drill/diamonddrill
```
Comment (Cyberboss) [inline code/game/mecha/working/ripley.dm]: here too

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**